### PR TITLE
feat(bucket): unique field option

### PR DESCRIPTION
**Pull request by htuna07**

### CODE REVIEW

**Comment by thesayyn** [comment]
Can we hold this PR until we land bucket rules? @htuna07.

There will be enormous conflicts between these PRs.

File: stacks/api/bucket/services/bucket.service.ts
Comment (thesayyn) [inline stacks/api/bucket/services/bucket.service.ts:28]: You can use getBucketDataCollection  from "@spica-server/bucket/services"

File: stacks/api/bucket/services/bucket.service.ts
Comment (thesayyn) [inline stacks/api/bucket/services/bucket.service.ts:27]: Also, you could use  super.insertOne().

File: stacks/api/bucket/src/crud.ts
Comment (thesayyn) [inline stacks/api/bucket/src/crud.ts:207]: You could return a message like `value of the property .title should unique across all documents.`

File: stacks/api/bucket/src/crud.ts
Comment (thesayyn) [inline stacks/api/bucket/src/crud.ts:254]: Same as above.

File: stacks/api/bucket/src/crud.ts
Comment (thesayyn) [inline stacks/api/bucket/src/crud.ts:297]: same as above

**Review by thesayyn — CHANGES_REQUESTED** [review]

**Comment by thesayyn** [comment]
you should wait for the https://github.com/spica-engine/spica/pull/521 first.

**Comment by thesayyn** [comment]
ping @htuna07

**Review by akhnos — APPROVED** [review]

**Review by thesayyn — APPROVED** [review]
lgtm

**Pull request merged**
